CRM/Contact/BAO/Relationship.php

```diff
@@ -304,7 +304,7 @@ public static function add(&$params, $ids = array(), $contactId = NULL) {
     if (!empty($relationshipId)) {
       $existingTypeID = CRM_Core_DAO::getFieldValue('CRM_Contact_DAO_Relationship', $relationshipId, 'relationship_type_id');
       $existingTypeName = CRM_Core_DAO::getFieldValue('CRM_Contact_DAO_RelationshipType', $existingTypeID, 'name_b_a');
-      if (empty($params['is_current_employer']) && $existingTypeName == 'Employer of' && (empty($params['is_active'])  || $existingTypeID != $type)) {
+      if ($existingTypeName == 'Employer of' && (empty($params['is_active']) || $existingTypeID != $type)) {
```

Comment:
Don't we still need to have !$params['is_current_employer'] here to respect any flag from the form?

Reply:
Updated the PR which should respect both API and form params. A new function is added which aims to decide if we need to set the current employer for the relationship or not. Tagging it as wip until I do a manual testing for some good number of scenarios. As a quick testing on the contact summary form, it worked fine for me.